packages/integrations/src/opnsense/opnsense-integration.ts

```diff
@@ -0,0 +1,207 @@
+import { fetchWithTrustedCertificatesAsync } from "@homarr/certificates/server";
+import { ParseError } from "@homarr/common/server";
+
+import { createChannelEventHistory } from "../../../redis/src/lib/channel";
+import { HandleIntegrationErrors } from "../base/errors/decorator";
+import { integrationAxiosHttpErrorHandler } from "../base/errors/http";
+import type { IntegrationTestingInput } from "../base/integration";
+import { Integration } from "../base/integration";
+import { TestConnectionError } from "../base/test-connection/test-connection-error";
+import type { TestingResult } from "../base/test-connection/test-connection-service";
+import type { FirewallSummaryIntegration } from "../interfaces/firewall-summary/firewall-summary-integration";
+import type {
+  FirewallCpuSummary,
+  FirewallInterface,
+  FirewallInterfacesSummary,
+  FirewallMemorySummary,
+  FirewallVersionSummary,
+} from "../interfaces/firewall-summary/firewall-summary-types";
+import {
+  opnsenseCPUSchema,
+  opnsenseInterfacesSchema,
+  opnsenseMemorySchema,
+  opnsenseSystemSummarySchema,
+} from "./opnsense-types";
+
+@HandleIntegrationErrors([integrationAxiosHttpErrorHandler])
+export class OPNsenseIntegration extends Integration implements FirewallSummaryIntegration {
+  protected async testingAsync(input: IntegrationTestingInput): Promise<TestingResult> {
+    const response = await input.fetchAsync(this.url("/api/diagnostics/system/system_information"), {
+      headers: {
+        Authorization: this.getAuthHeaders(),
+      },
+    });
+    if (!response.ok) return TestConnectionError.StatusResult(response);
+
+    const result = await response.json();
+    if (typeof result === "object" && result !== null) return { success: true };
+
+    return TestConnectionError.ParseResult(new ParseError("Expected object data"));
+  }
+
+  private getAuthHeaders() {
+    const username = super.getSecretValue("username");
+    const password = super.getSecretValue("password");
+    return "Basic " + btoa(`${username}:${password}`);
+  }
+
+  public async getFirewallVersionAsync(): Promise<FirewallVersionSummary> {
+    const responseSummary = await fetchWithTrustedCertificatesAsync(
+      this.url("/api/diagnostics/system/system_information"),
+      {
+        headers: {
+          Authorization: this.getAuthHeaders(),
+        },
+      },
+    );
+    if (!responseSummary.ok) {
+      throw new Error(
+        `Failed to fetch version for ${this.integration.name} (${this.integration.id}): ${responseSummary.statusText}`,
+      );
+    }
+    const summary = opnsenseSystemSummarySchema.safeParse(await responseSummary.json());
+    if (!summary.success) {
+      throw new Error(
+        `Failed to parse version for ${this.integration.name} (${this.integration.id}):\n${summary.error.message}`,
+      );
+    }
```

Comment:
Instead use `opnsenseSystemSummarySchema.parse` and error handling for parse error is handled by `@HandleIntegrationErrors` in base class